core/src/main/java/org/elasticsearch/index/query/PrefixQueryBuilder.java

```diff
@@ -75,7 +75,7 @@ public PrefixQueryBuilder queryName(String queryName) {
     @Override
     public void doXContent(XContentBuilder builder, Params params) throws IOException {
         builder.startObject(PrefixQueryParser.NAME);
-        if (boost == -1 && rewrite == null && queryName != null) {
+        if (boost == -1 && rewrite == null && queryName == null) {
```

Comment:
good point, I was not sure either, because the if was short enough and the query doesn't have too many fields. We will get rid of this if anyway in the query-refactoring branch soon :)